src/UI/PersonNewsFeedBar.jsx

```diff
@@ -10,6 +19,23 @@ const PreferencesForm = () => {
   });
   const [error, setError] = useState("");
 
+  let paperRef = useRef(null);
+
+  useEffect(() => {
```

Comment:
A general rule, try to stay away from useEffect when ever possible, btw, this was a very nice solution, still if a single prop can replace `useRef` and `useEffect` and attaching the event listeners, you should always take it

Not bad, but still the cleanest way is to use what the ui-library provides, ofc you couldn't know it all, no judgements here 😆 

So you can totally remove useRef and useEffect and do just `onClose={toggleModal}` like suggested below